src/views/router_views/enter-your-details/enter-your-details.njk

```diff
@@ -0,0 +1,146 @@
+{% extends "layouts/default.njk" %}
+
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "govuk/components/input/macro.njk" import govukInput %}
+{% from "govuk/components/select/macro.njk" import govukSelect %}
+
+{% block main_content %}
+{% set premisesField = ""%}
+{% set addressLine1Field = ""%}
+{% set townOrCityField = ""%}
+{% set countryField = ""%}
+{% set postCodeField = ""%}
+    {% if errors %}
+        {% for error in errors %}
+            {% if error.path === "premises" %}
+                {% set premisesField = {
+                    text: error.msg
+                }%}
+            {% elseif error.path === "addressLine1" %}
+                {% set addressLine1Field = {
+                    text: error.msg
+                }%}
+            {% elseif error.path === "addressLine2" %}
+                {% set addressLine2Field = {
+                    text: error.msg
+                }%}
+            {% elseif error.path === "townOrCity" %}
+                {% set townOrCityField = {
+                    text: error.msg
+                }%}
+            {% elseif error.path === "county" %}
+                {% set countyField = {
+                    text: error.msg
+                }%}
+            {% elseif error.path === "country" %}
+                {% set countryField = {
+                    text: error.msg
+                }%}
+            {% elseif error.path === "postCode" %}
+                {% set postCodeField = {
+                    text: error.msg
+                }%}
+            {% endif %}
+        {% endfor %}
+        {% else %}
+            {% set errorMessage = undefined %}
+    {% endif %}
```

Comment:
This is very verbose which makes it hard to erad and extend.
The format of the errors object should match that required by the error summary as stated above. 
This would allow of the error message to be populated directly at the macro like so
```
        errorMessage: errors['fieldName'] ? { text: errors['fieldName'].summary } : undefined
```